Code/Robot/Robot/Robot.ino

```diff
@@ -14,31 +14,71 @@
 #include <ArduinoSTL.h>
 #include <ArduinoJson.h>
 #include <StreamUtils.h>
-//#include <DualTB9051FTGMotorShieldBarebones.h>
 #include <DualTB9051FTGMotorShieldMod3230.h>
 #include <L298NMotorDriverMega.h>
+#include <QTRSensors.h>
 #include <queue>
 #include "Wheelbase.h"
+// Custom versions of libraries to debug (kinda useless)
 //#include "L298NMotorDriver.h"
-
+//#include <DualTB9051FTGMotorShieldBarebones.h>
 
 // Global variables :(
 const int cNumberOfWheels = 4;
+const uint8_t SensorCount = 8;
+uint16_t sensorValues[SensorCount];
+const int distPin1 = A4;              // Left IR rangefinder sensor
+const int distPin2 = A5;              // Right IR rangefinder sensor
+const int topLimitSwitchPin = 53;     // Replace XX with the actual pin number
+const int bottomLimitSwitchPin = 52;  // Replace YY with the actual pin number
 
 DualTB9051FTGMotorShieldMod3230 gMecanumMotors;
-L298NMotorDriverMega gSmolMotors(5, 32, 33, 6, 34, 35);
-//L298NMotorDriver small_motors(34,32,33,35,5,6);
-
+L298NMotorDriverMega gL2Motors(5, 34, 32, 6, 33, 35);
 Wheelbase* gWheelbase = new Wheelbase(5.0625, 4.386, 2.559);
+QTRSensors qtr;
 
 // Structs and enums
+enum MoveType {
+  eFreeDrive = 0,
+  eLineFollow = 1,
+  eScissor = 2,
+  eBelt = 3,
+  eCalibrate = 4,
+};
+
+// Onion for move-specific parameters
+union MoveParameters {
+  struct {
+    unsigned short direction;  // which way to drive (will be [x,y,theta] in the future)
+    unsigned long duration;    // how far to go (will be distance not time in the future)
+  } freedriveParams;           // For eFreeDrive
+
```

Comment:
```suggestion
  struct {
    Directions direction;  // which way to drive (will be [x,y,theta] in the future)
    unsigned long duration;    // how far to go (will be distance not time in the future)
  } freedriveParams;           // For eFreeDrive

```

Store value in the struct as the enum

Reply:
```Directions direction```
I just tried this. Wouldn't compile. Could you take a look at it?

Reply:
I'm sure that won't instantly compile until some of the other changes relating to freedrive.direction aee implemented